include/RAJA/exec-cuda/MemUtils_CUDA.hxx

```diff
@@ -62,37 +62,90 @@
 
 namespace RAJA
 {
+/// Maximum number of blocks that RAJA will launch
+/// threads are reused to finish all work
+#define RAJA_CUDA_MAX_NUM_BLOCKS (1024 * 16)
```

Comment:
A lot of what you're requesting stems from the fact that I've weaved in Jason's work, and we still need to do some cleanup/clarification and consistency checks.